dotnet/targets/Xamarin.Shared.Sdk.props

```diff
@@ -8,8 +8,15 @@
 
 	<Import Project="Xamarin.Shared.Sdk.TargetFrameworkInference.props" />
 	
-	<!-- Imports the .user.env file if exists and the build is from VS -->
-	<Import Project="$(MSBuildProjectFullPath).user.env" Condition="Exists('$(MSBuildProjectFullPath).user.env') And '$(BuildingInsideVisualStudio)' == 'true'" />
+	<PropertyGroup>
+		<!-- e.g: C:\Users\user1\app1\app1\obj\ -->
+		<_MobilePropsDir>$([System.IO.Path]::GetDirectoryName($(MSBuildProjectFullPath)))\$(BaseIntermediateOutputPath)</_MobilePropsDir>
```

Comment:
I believe `BaseIntermediateOutputPath` can be a full path, in which case this logic would fail.

Ref: https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022

Reply:
I just improved it based on this possible case